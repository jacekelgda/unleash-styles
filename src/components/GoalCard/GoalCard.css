/* GoalCard.css */

@media screen and (min-width: 1024px) {
	.GoalCard {
		width: calc(50% - 9px);
		align-self: flex-end;
    list-style: none;
	}
	.GoalCard:nth-child(even) {
		align-self: flex-start;
	}
}

.GoalCardIntro {
	margin-top: 40px;
	padding: 28px 18px 18px;
	border: 1px solid #e9edee;
	position: relative;
}

@media screen and (min-width: 768px) {
	.GoalCardIntro {
		margin: 30px;
		padding: 35px 30px 20px;
	}
	.GoalCardIntro:first-child {
		margin-top: 60px;
	}
}

.GoalCardIntro::before {
	position: absolute;
	content: '';
	border-radius: 50%;
	width: 18px;
	height: 18px;
	border: 2px solid #e9edee;
	background-color: #fff;
	top: -12px;
	left: calc(50% - 10px);
}
.GoalCardIntro--full::before {
	content: "✓";
	text-align: center;
	color: #fff;
	background-color: #e9edee;
}

@media screen and (min-width: 768px) {
	.GoalCardIntro::before {
		left: -50px;
		top: calc(50% - 10px);
	}
}
@media screen and (min-width: 1024px) {
	.GoalCard:nth-child(odd) .GoalCardIntro::before {
		left: -54px;
		top: calc(50% - 10px);
	}
	.GoalCard:nth-child(even) .GoalCardIntro::before {
		left: auto;
		right: -48px;
		top: calc(50% - 10px);
	}
}

/* top right corner dots */
.GoalCardIntro::after {
	position: absolute;
	content: url(./images/three-dots.svg);
	border-radius: 50%;
	width: 20px;
	height: 20px;
	top: 25px;
	right: 25px;
	cursor: pointer;
	display: block;
}
@media screen and (min-width: 768px) {
	.GoalCardIntro::after {
		top: 32px;
		right: 35px;
	}
}

.GoalCardTitle {
	font: 700 22px/1 'Montserrat', sans-serif;
	text-transform: uppercase;
	color: #34323b;
	margin: 0;
}
@media screen and (max-width: 767px) {
	.GoalCardTitle {
		max-width: 60%;
	}
}
@media screen and (min-width: 768px) {
	.GoalCardTitle {
		font-size: 24px;
	}
}

.GoalCardDescription {
	font-size: 14px;
	line-height: 20px;
	color: #34323b;
	margin: 15px 0;
}

.GoalCardProgress {
	display: flex;
	justify-content: space-between;
	align-items: center;
}

.GoalCardSteps {
	display: inline-block;
	font: 14px 'Montserrat', sans-serif;
	text-transform: uppercase;
	color: #9a9d9e;
	min-width: 90px;
}
.GoalCardStepsLabel {
	display: inline-block;
	margin-right: 2px;
}
.GoalCardStepsBar {
	margin: 0 -1px;
	display: inline-block;
}

@media screen and (max-width: 767px) {
	.GoalCardProgressBar {
		display: none;
	}
}
@media screen and (min-width: 768px) {
	.GoalCardProgressBar {
		flex-grow: 1;
		margin: 0 20px;
		height: 6px;
		border-radius: 8px;
		background-color: #e1e1e1;
	}
	.GoalCardProgressBar::-webkit-progress-bar {
		height: 6px;
		border-radius: 8px;
		background-color: #e1e1e1;
	}
	.GoalCardProgressBar::-moz-progress-bar {
		height: 6px;
		border-radius: 8px;
		background-color: #e1e1e1;
	}
	.GoalCardProgressBar::-webkit-progress-value {
		border-radius: 8px 0 0 8px;
		height: 6px;
		background-color: #5fcf21;
	}
	.GoalCardProgressBar::-moz-progress-value {
		border-radius: 8px 0 0 8px;
		height: 6px;
		background-color: #5fcf21;
	}
	.GoalCardProgressBar::-ms-fill {
		border-radius: 8px 0 0 8px;
		height: 6px;
		background-color: #5fcf21;
	}

	.GoalCardProgressBar--red::-webkit-progress-value {
		background-color: red;
	}
	.GoalCardProgressBar--red::-moz-progress-value {
		background-color: red;
	}
	.GoalCardProgressBar--red::-ms-fill {
		background-color: red;
	}
}


.GoalCardSwitchView {
	display: inline-flex;
	align-items: center;
	justify-content: space-between;
	font: 14px 'Montserrat', sans-serif;
	text-transform: uppercase;
	color: #9a9d9e;
	padding: 5px;
	border: none;
	cursor: pointer;
	position: relative;
	background-color: transparent;
	min-width: 130px;
}
.GoalCardSwitchView::after {
	display: inline-block;
	content: url(./images/arrow--up.svg);
	transform: rotate(180deg);
	width: 15px;
	height: 20px;
	margin-left: 10px;
	/*transition: transform .3s ease-out;*/
}
.GoalCardSwitchView::before {
	line-height: 20px;
	content: 'Show more';
	animation-duration: .5s;
	animation-name: fade;
	animation-iteration-count: 1;
}
.GoalCardSwitchView.is-open::after {
	transform: rotate(0deg);
	align-self: baseline;
}
.GoalCardSwitchView.is-open::before {
	content: 'Show less';
	animation-duration: .5s;
	animation-name: fadeIn;
	animation-iteration-count: 1;
}

@keyframes fadeIn {
	from {
		opacity: .5;
	}
	to {
		opacity: 1;
	}
}
@keyframes fade {
	from {
		opacity: .5;
	}
	to {
		opacity: 1;
	}
}

/* GoalCardChecklist.css */

.GoalCardInfo {
	display: none;
}
.GoalCard.is-open .GoalCardInfo {
	display: block;
}

/* Desktop layout */
@media screen and (min-width: 1024px) {
	.GoalCardInfo {
		margin-left: 60px;
	}
}

.GoalCardChecklist {
	position: relative;
	list-style: none;
	margin: 30px 0 0 20px;
	padding: 0;
}
@media screen and (min-width: 768px) {
	.GoalCardChecklist {
		margin-left: 60px;
		max-width: calc(100% - 120px);
	}
}
@media screen and (min-width: 1024px) {
	.GoalCardChecklist {
		margin: 0;
		max-width: 100%;
	}
}
.GoalCardChecklist:before {
	content: '';
	position: absolute;
	left: 0;
	top: -30px;
	width: 2px;
	height: calc(100% - 25px);
	background-color: #e9edee;
}

.GoalCardItemCheckbox {
	margin: 0 0 25px;
	position: relative;
	display: -webkit-flex;
	display: -moz-flex;
	display: -ms-flex;
	display: -o-flex;
	display: flex;
	-ms-align-items: center;
	align-items: center;
}
@media screen and (min-width: 1024px) {
	.GoalCardItemCheckbox {
		max-width: calc(100% - 60px);
	}
}

.GoalCardItemCheckbox::after {
	position: absolute;
	content: url(./images/three-dots.svg);
	border-radius: 50%;
	width: 20px;
	height: 4px;
	top: 0;
	right: 0;
}
@media screen and (min-width: 1024px) {
	.GoalCardItemCheckbox::after {
		transition: opacity .3s ease-out;
		opacity: 0;
		visibility: hidden;
	}
	.GoalCardItemCheckbox:hover::after {
		opacity: 1;
		visibility: visible;
	}
}

.GoalCardItemInput {
	display: inline-block;
	vertical-align: middle;
	width: 20px;
	height: 20px;
	flex: 0 0 20px;
	position: relative;
}

/* checkbox styling */
.GoalCardItemInput::after {
	display: inline-block;
	content: '';
	position: absolute;
	left: -18px; top: -7px;
	width: 30px;
	height: 30px;
	border: 2px solid #e9edee;
	background-color: #fff;
	transition: background-color .2s ease-out;
}
.GoalCardItemCheckbox:hover .GoalCardItemInput::after,
.GoalCardItemInput:checked::after {
	content: '✓';
	text-align: center;
	line-height: 30px;
	font-size: 20px;
	vertical-align: middle;
}

.GoalCardItemCheckbox:hover .GoalCardItemInput:not(:checked)::after {
	color: #5fcf21;
}
.GoalCardItemInput:checked::after {
	border-color: #5fcf21;
	background-color: #5fcf21;
	color: #fff;
}
.GoalCardItemInput:checked + .GoalCardItemLabel {
	text-decoration: line-through;
	color: #b8b7bc;
}

.GoalCardItemLabel,
.GoalCardItemCheckbox__edit {
	display: block;
	font-size: 16px;
	line-height: 1;
	font-weight: 600;
	color: #34323b;
	padding-right: 30px;
	margin-left: 5px;
}
.GoalCardItemCheckbox__edit {
	font-family: 'Open Sans', sans-serif;
	line-height: 30px;
	border: 2px solid #e9edee;
	padding: 0 30px 0 5px;
	width: 100%;
}

/* disabled checkboxes */
.GoalCardItemCheckbox:hover .GoalCardItemInput:disabled::after,
.GoalCardItemInput:disabled::after {
	content: '';
}


/* add a new step button */
.GoalCardAddNew.is-hidden {
	display: none;
}
.GoalCardAddNew {
	margin-left: -8px;
}
.GoalCardAddNewButton {
	font: 14px/1 'Montserrat', sans-serif;
	text-transform: uppercase;
	color: #f33c57;
	border: none;
	padding: 5px;
	background-color: transparent;
	cursor: pointer;
}
.GoalCardAddNewButtonIcon {
	display: inline-block;
	vertical-align: middle;
	font-size: 22px;
	margin-right: 4px;
}
