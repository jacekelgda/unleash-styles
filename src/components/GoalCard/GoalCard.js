import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './GoalCard.css'

/**
 * GoalCard
 */
class GoalCard extends Component {
  render () {
    const { title, onClick } = this.props
    const goalItemsCount = this.props.children.length
    const goalItemsCompletedCount = this.props.children.filter((item) => (item.props.isComplete)).length

    return (
      <li className={styles.GoalCard}>
        <div className={styles.GoalCardIntro}>
          <h3 className={styles.GoalCardTitle}>{title}</h3>
          <p className={styles.GoalCardDescription} />
          <div className={styles.GoalCardProgress}>
            <div className={styles.GoalCardSteps}>
              <span className={styles.GoalCardStepsLabel}>Steps:</span>
              <span>{goalItemsCompletedCount}</span>
              <span className={styles.GoalCardStepsBar}>/</span>
              <span>{goalItemsCount}</span>
            </div>
            <progress className={styles.GoalCardProgressBar} max={goalItemsCount} value={goalItemsCompletedCount} />
            <button className={styles.GoalCardSwitchView} />
          </div>
        </div>
        <div className={styles.GoalCardInfo}>
          <ul className={styles.GoalCardChecklist}>
            {this.props.children}
            <li className={styles.GoalCardAddNew}>
              <button className={styles.GoalCardAddNewButton} onClick={onClick}>
                <span className={styles.GoalCardAddNewButtonIcon}>+</span> add a new step
              </button>
            </li>
          </ul>
        </div>
      </li>
    )
  }
}

GoalCard.propTypes = {
  /** Title of the GoalCard */
  title: PropTypes.string.isRequired,
  /** Function to be triggered when adding a new item */
  onClick: PropTypes.func.isRequired
}

export default GoalCard
