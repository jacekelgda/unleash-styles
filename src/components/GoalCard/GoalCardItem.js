/** @flow */
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './GoalCard.css'

/**
 * GoalCardItem
 */
class GoalCardItem extends Component {
  render () {
    const { label, isComplete } = this.props

    return (
      <li className={styles.GoalCardItemCheckbox}>
        <input className={styles.GoalCardItemInput} type='checkbox' id={label} checked={isComplete} />
        <label className={styles.GoalCardItemLabel} for={label}>
          {label}
        </label>
      </li>
    )
  }
}

GoalCardItem.propTypes = {
  /** label of the GoalCardItem */
  label: PropTypes.string.isRequired,
  /** wether the item is completed */
  isComplete: PropTypes.bool
}

export default GoalCardItem
