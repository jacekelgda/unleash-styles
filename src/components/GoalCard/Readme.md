GoalCard component:

    <GoalCard title='Lorem Ipsum' onClick={() => {}}>
      <GoalCardItem label='test1' isComplete />
      <GoalCardItem label='test2' isComplete />
      <GoalCardItem label='test3' />
      <GoalCardItem label='test4' />
    </GoalCard>

GoalCardItem component:

  <GoalCardItem label='test1' isComplete />
