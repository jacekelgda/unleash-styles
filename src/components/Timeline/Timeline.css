.Timeline_header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 40px 0;
  position: relative;
}
@media screen and (min-width: 768px) {
  .Timeline_header::after {
    display: block;
    position: absolute;
    left: 30px;
    top: calc(50% - 1px);
    content: '';
    width: calc(100% - 60px);
    height: 1px;
    background-color: #e9e9e9;
    z-index: 0;
  }
}

.Timeline_search {
  position: relative;
  padding-left: 16px;
  background-color: #fff;
  width: 8rem;
  z-index: 1;
}
.Timeline_search::before {
  display: block;
  position: absolute;
  left: 0;
  top: calc(50% - 9px);
  content: url(../../assets/search.svg);
  width: 16px;
  height: 15px;
}

.Timeline_search_input {
  font: 14px/2 'Montserrat', sans-serif;
  text-transform: uppercase;
  background-color: transparent;
  color: #9a9d9e;
  border: none;
  padding: 0 11px;
  width: 8rem;
}
.Timeline_search_input::placeholder {
  color: #9a9d9e;
}

.Timeline_items {
  list-style: none;
  margin: 0;
  padding: 0 18px;
  position: relative;
}
@media screen and (min-width: 768px) {
  .Timeline_items {
    padding: 0 30px;
    margin-left: 65px;
  }
  .Timeline_items::before {
    content: '';
    position: absolute;
    left: 21px;
    top: -60px;
    width: 2px;
    height: calc(100% + 60px);
    background-color: #e9edee;
  }
}
@media screen and (min-width: 1024px) {
  .Timeline_items {
    display: flex;
    flex-direction: column;
    padding: 0;
    margin: 0;
  }
  .Timeline_items::before {
    top: -36px;
    left: calc(50% - 4px);
  }
}

.Timeline_date_label {
  margin: 0 18px;
  font: 14px/34px 'Montserrat', sans-serif;
  text-transform: uppercase;
  text-align: center;
  color: #34323b;
  background-color: #e1e1e1;
}
@media screen and (min-width: 768px) {
  .Timeline_date_label {
    margin: 0 30px;
    width: 115px;
  }
}
@media screen and (min-width: 1024px) {
  .Timeline_date_label {
    margin: 0;
    position: absolute;
    left: calc(50% - 62px);
    margin-top: -70px;
  }
}
