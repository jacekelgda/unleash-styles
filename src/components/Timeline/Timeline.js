/** @flow */
import React, { Component } from 'react'
import styles from './Timeline.css'

/**
 * Timeline component creates visible timeline for it's children
 */
class Timeline extends Component {
  render () {
    return (
      <section className={styles.Timeline}>

        <header className={styles.Timeline_header}>
          <div className={styles.Timeline_search}>
            <input className={styles.Timeline_search_input} type='search' placeholder='Search' />
          </div>
        </header>

        <div className={styles.Timeline_date_label}>
          Today
        </div>

        <ol className={styles.Timeline_items}>
          {this.props.children}
        </ol>
      </section>
    )
  }
}

export default Timeline
