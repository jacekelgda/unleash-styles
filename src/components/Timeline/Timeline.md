Timeline composition example:

    <Timeline>
      <GoalCard title='Lorem Ipsum' onClick={() => {}}>
        <GoalCardItem label='test1' isComplete />
        <GoalCardItem label='test2' isComplete />
        <GoalCardItem label='test3' />
        <GoalCardItem label='test4' />
      </GoalCard>
      <GoalCard title='Lorem Ipsum' onClick={() => {}}>
        <GoalCardItem label='test1' isComplete />
        <GoalCardItem label='test2' isComplete />
        <GoalCardItem label='test3' />
        <GoalCardItem label='test4' />
      </GoalCard>
      <GoalCard title='Lorem Ipsum' onClick={() => {}}>
        <GoalCardItem label='test1' isComplete />
        <GoalCardItem label='test2' isComplete />
        <GoalCardItem label='test3' />
        <GoalCardItem label='test4' />
      </GoalCard>
    </Timeline>
