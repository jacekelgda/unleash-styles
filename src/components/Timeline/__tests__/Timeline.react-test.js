/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'
import Timeline from '../Timeline'
import GoalCard from '../../GoalCard/GoalCard'
import GoalCardItem from '../../GoalCard/GoalCardItem'

test('Timeline renders correctly', () => {
  const wrapper = shallow(<Timeline />)
  expect(wrapper).toMatchSnapshot()
})

test('Timeline renders its children correctly', () => {
  const wrapper = shallow(
    <Timeline>
      <GoalCard title='Lorem Ipsum' onClick={() => {}}>
        <GoalCardItem label='test1' isComplete />
        <GoalCardItem label='test2' />
      </GoalCard>
      <GoalCard title='Lorem Ipsum' onClick={() => {}}>
        <GoalCardItem label='test1' isComplete />
        <GoalCardItem label='test2' />
      </GoalCard>
      <GoalCard title='Lorem Ipsum' onClick={() => {}}>
        <GoalCardItem label='test1' isComplete />
        <GoalCardItem label='test2' />
      </GoalCard>
    </Timeline>
  )
  expect(wrapper).toMatchSnapshot()
})
