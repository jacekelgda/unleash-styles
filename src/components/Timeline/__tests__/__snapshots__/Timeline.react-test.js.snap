// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="Timeline"
>
    <header
        className="Timeline_header"
    >
        <div
            className="Timeline_search"
        >
            <input
                className="Timeline_search_input"
                placeholder="Search"
                type="search"
            />
        </div>
    </header>
    <div
        className="Timeline_date_label"
    >
        Today
    </div>
    <ol
        className="Timeline_items"
    />
</section>,
  "nodes": Array [
    <section
      className="Timeline"
>
      <header
            className="Timeline_header"
      >
            <div
                  className="Timeline_search"
            >
                  <input
                        className="Timeline_search_input"
                        placeholder="Search"
                        type="search"
                  />
            </div>
      </header>
      <div
            className="Timeline_date_label"
      >
            Today
      </div>
      <ol
            className="Timeline_items"
      />
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Timeline />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Timeline {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="Timeline"
>
          <header
                    className="Timeline_header"
          >
                    <div
                              className="Timeline_search"
                    >
                              <input
                                        className="Timeline_search_input"
                                        placeholder="Search"
                                        type="search"
                              />
                    </div>
          </header>
          <div
                    className="Timeline_date_label"
          >
                    Today
          </div>
          <ol
                    className="Timeline_items"
          />
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="Timeline"
>
          <header
                    className="Timeline_header"
          >
                    <div
                              className="Timeline_search"
                    >
                              <input
                                        className="Timeline_search_input"
                                        placeholder="Search"
                                        type="search"
                              />
                    </div>
          </header>
          <div
                    className="Timeline_date_label"
          >
                    Today
          </div>
          <ol
                    className="Timeline_items"
          />
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Timeline />,
}
`;

exports[`Timeline renders its children correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="Timeline"
>
    <header
        className="Timeline_header"
    >
        <div
            className="Timeline_search"
        >
            <input
                className="Timeline_search_input"
                placeholder="Search"
                type="search"
            />
        </div>
    </header>
    <div
        className="Timeline_date_label"
    >
        Today
    </div>
    <ol
        className="Timeline_items"
    >
        <GoalCard
            onClick={[Function]}
            title="Lorem Ipsum"
        >
            <GoalCardItem
                isComplete={true}
                label="test1"
            />
            <GoalCardItem
                label="test2"
            />
        </GoalCard>
        <GoalCard
            onClick={[Function]}
            title="Lorem Ipsum"
        >
            <GoalCardItem
                isComplete={true}
                label="test1"
            />
            <GoalCardItem
                label="test2"
            />
        </GoalCard>
        <GoalCard
            onClick={[Function]}
            title="Lorem Ipsum"
        >
            <GoalCardItem
                isComplete={true}
                label="test1"
            />
            <GoalCardItem
                label="test2"
            />
        </GoalCard>
    </ol>
</section>,
  "nodes": Array [
    <section
      className="Timeline"
>
      <header
            className="Timeline_header"
      >
            <div
                  className="Timeline_search"
            >
                  <input
                        className="Timeline_search_input"
                        placeholder="Search"
                        type="search"
                  />
            </div>
      </header>
      <div
            className="Timeline_date_label"
      >
            Today
      </div>
      <ol
            className="Timeline_items"
      >
            <GoalCard
                  onClick={[Function]}
                  title="Lorem Ipsum"
            >
                  <GoalCardItem
                        isComplete={true}
                        label="test1"
                  />
                  <GoalCardItem
                        label="test2"
                  />
            </GoalCard>
            <GoalCard
                  onClick={[Function]}
                  title="Lorem Ipsum"
            >
                  <GoalCardItem
                        isComplete={true}
                        label="test1"
                  />
                  <GoalCardItem
                        label="test2"
                  />
            </GoalCard>
            <GoalCard
                  onClick={[Function]}
                  title="Lorem Ipsum"
            >
                  <GoalCardItem
                        isComplete={true}
                        label="test1"
                  />
                  <GoalCardItem
                        label="test2"
                  />
            </GoalCard>
      </ol>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Timeline>
        <GoalCard
                onClick={[Function]}
                title="Lorem Ipsum"
        >
                <GoalCardItem
                        isComplete={true}
                        label="test1"
                />
                <GoalCardItem
                        label="test2"
                />
        </GoalCard>
        <GoalCard
                onClick={[Function]}
                title="Lorem Ipsum"
        >
                <GoalCardItem
                        isComplete={true}
                        label="test1"
                />
                <GoalCardItem
                        label="test2"
                />
        </GoalCard>
        <GoalCard
                onClick={[Function]}
                title="Lorem Ipsum"
        >
                <GoalCardItem
                        isComplete={true}
                        label="test1"
                />
                <GoalCardItem
                        label="test2"
                />
        </GoalCard>
</Timeline>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Timeline {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <GoalCard
              onClick={[Function]}
              title="Lorem Ipsum"
>
              <GoalCardItem
                            isComplete={true}
                            label="test1"
              />
              <GoalCardItem
                            label="test2"
              />
</GoalCard>,
            <GoalCard
              onClick={[Function]}
              title="Lorem Ipsum"
>
              <GoalCardItem
                            isComplete={true}
                            label="test1"
              />
              <GoalCardItem
                            label="test2"
              />
</GoalCard>,
            <GoalCard
              onClick={[Function]}
              title="Lorem Ipsum"
>
              <GoalCardItem
                            isComplete={true}
                            label="test1"
              />
              <GoalCardItem
                            label="test2"
              />
</GoalCard>,
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="Timeline"
>
          <header
                    className="Timeline_header"
          >
                    <div
                              className="Timeline_search"
                    >
                              <input
                                        className="Timeline_search_input"
                                        placeholder="Search"
                                        type="search"
                              />
                    </div>
          </header>
          <div
                    className="Timeline_date_label"
          >
                    Today
          </div>
          <ol
                    className="Timeline_items"
          >
                    <GoalCard
                              onClick={[Function]}
                              title="Lorem Ipsum"
                    >
                              <GoalCardItem
                                        isComplete={true}
                                        label="test1"
                              />
                              <GoalCardItem
                                        label="test2"
                              />
                    </GoalCard>
                    <GoalCard
                              onClick={[Function]}
                              title="Lorem Ipsum"
                    >
                              <GoalCardItem
                                        isComplete={true}
                                        label="test1"
                              />
                              <GoalCardItem
                                        label="test2"
                              />
                    </GoalCard>
                    <GoalCard
                              onClick={[Function]}
                              title="Lorem Ipsum"
                    >
                              <GoalCardItem
                                        isComplete={true}
                                        label="test1"
                              />
                              <GoalCardItem
                                        label="test2"
                              />
                    </GoalCard>
          </ol>
</section>,
        "_debugID": 4,
        "_renderedOutput": <section
          className="Timeline"
>
          <header
                    className="Timeline_header"
          >
                    <div
                              className="Timeline_search"
                    >
                              <input
                                        className="Timeline_search_input"
                                        placeholder="Search"
                                        type="search"
                              />
                    </div>
          </header>
          <div
                    className="Timeline_date_label"
          >
                    Today
          </div>
          <ol
                    className="Timeline_items"
          >
                    <GoalCard
                              onClick={[Function]}
                              title="Lorem Ipsum"
                    >
                              <GoalCardItem
                                        isComplete={true}
                                        label="test1"
                              />
                              <GoalCardItem
                                        label="test2"
                              />
                    </GoalCard>
                    <GoalCard
                              onClick={[Function]}
                              title="Lorem Ipsum"
                    >
                              <GoalCardItem
                                        isComplete={true}
                                        label="test1"
                              />
                              <GoalCardItem
                                        label="test2"
                              />
                    </GoalCard>
                    <GoalCard
                              onClick={[Function]}
                              title="Lorem Ipsum"
                    >
                              <GoalCardItem
                                        isComplete={true}
                                        label="test1"
                              />
                              <GoalCardItem
                                        label="test2"
                              />
                    </GoalCard>
          </ol>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Timeline>
    <GoalCard
        onClick={[Function]}
        title="Lorem Ipsum"
    >
        <GoalCardItem
            isComplete={true}
            label="test1"
        />
        <GoalCardItem
            label="test2"
        />
    </GoalCard>
    <GoalCard
        onClick={[Function]}
        title="Lorem Ipsum"
    >
        <GoalCardItem
            isComplete={true}
            label="test1"
        />
        <GoalCardItem
            label="test2"
        />
    </GoalCard>
    <GoalCard
        onClick={[Function]}
        title="Lorem Ipsum"
    >
        <GoalCardItem
            isComplete={true}
            label="test1"
        />
        <GoalCardItem
            label="test2"
        />
    </GoalCard>
</Timeline>,
}
`;
